lib/modules/hosts/vidly.js

```diff
@@ -0,0 +1,52 @@
+modules['showImages'].siteModules['vidly'] = {
+    domains: ['vidly.io'],
+    matchRE: /^https?:\/\/(?:www\.)?vidly\.io\/([\w]+)/i,
+    detect: function (href, elem) {
+        return href.indexOf('vidly.io') !== -1 || href.indexOf('https://vidly.io') !== -1;
+    },
+    handleLink: function (elem) {
+        var def = $.Deferred();
+        var hashRe = /^https?:\/\/(?:www\.)?vidly\.io\/([\w]+)/i;
+        var groups = hashRe.exec(elem.href);
+        if (groups) {           
+            var input = groups['input'];
+            var splinput = input.split('/');
+            var apiURL = 'https://vidly.io/media/' + splinput[splinput.length - 1];
+            RESUtils.runtime.ajax({
+                method: 'GET',
+                url: apiURL,
+                // aggressiveCache: true,
+                onload: function (response) {
+                    try {
+                        var json = JSON.parse(response.responseText);
+                        if (json && json.length > 0 && json[0]['outputs'] && json[0]['outputs']) {
```

Comment:
Yeah, that's just a separate thing I noticed - if it made a difference I'd be baffled.

That's the generic error logged when the promise returned by `handleLink()` is rejected; if you reject with a value, the value will be logged as well.

The error occurs, as mentioned, because on line 23, it's trying to `JSON.parse` a string containing a URL (i.e. not a proper JSON object).
In this case, I don't think you need to parse anything else after parsing the response; you should be able to use the URL directly from `json[0]['outputs']`.
![capture](https://cloud.githubusercontent.com/assets/7673145/9923467/4f250f48-5cc3-11e5-8b4b-2ad773792de4.PNG)